src/platform/plugins/shared/discover/public/application/main/state_management/discover_data_state_container.ts

```diff
@@ -398,6 +416,11 @@ export function getDataStateContainer({
                 );
               }
 
+              const { resetDefaultProfileState: currentResetDefaultProfileState } = getCurrentTab();
+              if (currentResetDefaultProfileState.resetId !== resetDefaultProfileState.resetId) {
+                return;
+              }
```

Comment:
I think we need to keep these lines above since `postFetchStateUpdate` shouldn't run in `resetId` doesn't match.